src/apps/loan-list/list/loan-list.tsx

```diff
@@ -0,0 +1,159 @@
+import MaterialDecorator from "../materials/material-decorator";
+import { useText } from "../../../core/utils/text";
+import DueDateLoansModal from "../modal/due-date-loans-modal";
+
+const LoanList: React.FC = () => {
+  const t = useText();
+  const [loans, setLoans] = useState<LoanV2[] | null>([]);
+  const [dueDates, setDueDates] = useState<string[]>();
+  const [dueDateModal, setDueDateModal] = useState<string>("");
+  const [loansModal, setLoansModal] = useState<LoanV2[] | null>();
+  const [renewable, setRenewable] = useState<number | null>(null);
+  const [amountOfLoans, setAmountOfLoans] = useState<number>(0);
+  const [view, setView] = useState<string>("list");
+
+  useEffect(() => {
+    dayjs.extend(localizedFormat);
+  }, []);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    // regex for finding date string from modal query param
+    const regex = /^\d{4}-\d{2}-\d{2}$/;
+    // modal query param
+    const modalString = searchParams.get("modal");
+
+    if (modalString && loans) {
+      const found = modalString.toString().match(regex);
+      // If there is a date string in the modal query param
+      if (found) {
+        setDueDateModal(found[0]);
+        // THe loans are filtered with said date string
+        const loansForModal = loans.filter(
+          ({ loanDetails }) => loanDetails.dueDate === dueDateModal
+        );
+        // Amount of renewable loans are determined, used in the ui
+        const amountOfRenewableLoans = loansForModal.filter(
+          ({ isRenewable }) => isRenewable
+        ).length;
+
+        // Loans for modal (the modal shows loans stacked by due date)
+        setLoansModal(loansForModal);
+        setRenewable(amountOfRenewableLoans);
+      }
+    }
+  }, [loans, dueDateModal]);
+
+  useEffect(() => {
+    getLoansV2().then((value) => {
+      if (value) {
+        const listOfDueDates = value.map((a) => a.loanDetails.dueDate);
+        const uniqeListOfDueDates = Array.from(new Set(listOfDueDates));
+
+        // The due dates are used for the stacked materials
+        // The stacked materials view shows materials stacked by
+        // due date, and for this we need a uniqe list of due dates
+        setDueDates(uniqeListOfDueDates);
+
+        // Loans are sorted by due date
+        const sortedByDueDate = value.sort(
+          (objA, objB) =>
+            new Date(objA.loanDetails.dueDate).getTime() -
+            new Date(objB.loanDetails.dueDate).getTime()
+        );
+        setLoans(sortedByDueDate);
+        setAmountOfLoans(sortedByDueDate.length);
+      }
+    });
+  }, []);
+
+  function openModalDueDate(dueDateModalInput: string) {
+    if (loans) {
+      setDueDateModal(dueDateModalInput);
+      setLoansModal(
+        loans.filter(
+          ({ loanDetails }) => loanDetails.dueDate === dueDateModalInput
+        )
+      );
+    }
+  }
+
+  return (
+    <div aria-hidden={loansModal !== null ?? true}>
```

Comment:
Shouldn't this be a string?
Maybe:
```tsx
<div aria-hidden={loansModal ?? "true"}>
```
But according to [this](https://github.com/danskernesdigitalebibliotek/dpl-react/pull/55/files#diff-2cdd62020bea6d73b0ed5c5b7ead29e571bdb06dedcfeb649598dcc202fe42c3R17) loansModal is: `LoanV2[] | null`.
I am confused :)